ppreference/ExtrapolateDataDrivenReference.py

```diff
@@ -0,0 +1,127 @@
+'''
+python script for the pT extrapolation of data-driven pp references
+run: python ExtrapolateDataDrivenCrossSection.py configFile.yml
+'''
+
+import sys
+import argparse
+import numpy as np
+import yaml
+from ROOT import TFile, TF1, TCanvas, TH1F, TLegend, TGraphAsymmErrors  # pylint: disable=import-error,no-name-in-module
+sys.path.append('..')
+from utils.AnalysisUtils import ComputeRatioDiffBins
+from utils.StyleFormatter import SetGlobalStyle, SetObjectStyle, GetROOTColor, GetROOTMarker
+
+SetGlobalStyle(padbottommargin=0.13)
+
+parser = argparse.ArgumentParser(description='Arguments to pass')
+parser.add_argument('cfgFileName', metavar='text', default='cfgFileName.yml',
+                    help='config file name with root input files')
+args = parser.parse_args()
+
+with open(args.cfgFileName, 'r') as ymlCutSetFile:
+    inCfg = yaml.load(ymlCutSetFile, yaml.FullLoader)
+
+inFile = TFile.Open(inCfg['infilename'])
+hCrossSec = inFile.Get('hCrossSection')
+hCrossSec.SetName('hCrossSecMeas')
+hCrossSec.SetDirectory(0)
+gCrossSec = inFile.Get('gCrossSectionSystTot')
+gCrossSec.SetName('gCrossSectionSystTotMeas')
+gCrossSecLumi = inFile.Get('gCrossSectionSystLumi')
+systErr = inFile.Get('AliHFSystErr')
+SetObjectStyle(hCrossSec, color=GetROOTColor('kRed+1'), markerstyle=GetROOTMarker('kFullSquare'), markersize=0.8)
+SetObjectStyle(gCrossSec, color=GetROOTColor('kRed+1'), fillstyle=0)
+inFile.Close()
+
+ptMaxMeas = hCrossSec.GetXaxis().GetXmax()
+ptLimsMeas = [hCrossSec.GetBinLowEdge(iPt) for iPt in range(1, hCrossSec.GetNbinsX()+1)]
+ptLimsMeas.append(ptMaxMeas)
+ptMinMeas = ptLimsMeas[0]
+
+colors = {'cent': GetROOTColor('kBlack'), 'min': GetROOTColor('kRed+1'), 'max': GetROOTColor('kAzure+4')}
+hFONLL, hDataOverFONLL, fDataOverFONLL = {}, {}, {}
+inFileFONLL = TFile.Open(inCfg['extrap']['FONLLfilename'])
+for var in ['cent', 'min', 'max']:
+    hFONLL[var] = inFileFONLL.Get(inCfg['extrap']['hist'][var])
+    hDataOverFONLL[var] = ComputeRatioDiffBins(hCrossSec, hFONLL[var])
+    hDataOverFONLL[var].Scale(1.e6)
+    SetObjectStyle(hDataOverFONLL[var], color=colors[var], fillstyle=0)
+    fDataOverFONLL[var] = TF1(f'fDataOverFONLL{var}', 'pol0', ptMinMeas, ptMaxMeas)
+    SetObjectStyle(fDataOverFONLL[var], color=colors[var])
+    hDataOverFONLL[var].Fit(fDataOverFONLL[var], '0')
```

Comment:
Don't we need to use both the statistical and systematic uncertainty for the data?